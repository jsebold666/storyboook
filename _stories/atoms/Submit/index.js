import React from 'react';
import { text, selectV2 as select, boolean, object } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import readme from './README.md';
import Submit from '../../../components/atoms/Submit';

const component = () => (
  <Submit onChange={action('changed')} value={text('value', 'Submit content')} />
);

export default [readme, component];
