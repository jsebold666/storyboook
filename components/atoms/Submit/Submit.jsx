import React, { Component } from 'react';
import styled from 'styled-components';
import shared from '../../consts/shared';
import { FormContext, uid } from '../Form/Form';

const Input = styled.input`
  min-width: 200px;
  width: 100%;
  height: 80px;
  border: 0;
  padding: 24px 16px 24px 24px;
  outline: none;
  background-color: ${shared['tealish']};
  color: ${shared['white']};

  &:hover:not(.btn-default):not(.btn-loading):not([disabled]) {
    background-color: #43e4e4;
  }
  &:active:not(.btn-default):not(.btn-loading):not([disabled]) {
    background-color: #2fbcbc;
  }

  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  letter-spacing: normal;

  &:disabled {
    background-color: ${shared['cloudy-blue']};
    &,
    & * {
      cursor: not-allowed;
    }
  }
`;

export default class Submit extends Component {
  render() {
    return <Input type="submit" value={this.props.value} />;
  }
}
