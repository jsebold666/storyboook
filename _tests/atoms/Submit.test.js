/* globals mount */
import React from 'react';
import Submit from '../../components/atoms/Submit';

describe('Expect <Submit>', () => {
  it('to render', () => {
    const props = {
      value: 'This is text',
      onChange: () => {}
    };

    const wrapper = mount(<Submit {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('to render properly with empty', () => {
    const props = {
      value: '',
      onChange: () => {}
    };

    const wrapper = mount(<Submit {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
