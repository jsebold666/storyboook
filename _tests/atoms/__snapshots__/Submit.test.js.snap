// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Submit> to render 1`] = `
<Submit
  onChange={[Function]}
  value="This is text"
>
  <Submit__Input
    type="submit"
    value="This is text"
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Submit__Input-sc-1h1p8pz-0",
            "isStatic": true,
            "lastClassName": "kMJEtZ",
            "rules": Array [
              "
  min-width: 200px;
  width: 100%;
  height: 80px;
  border: 0;
  padding: 24px 16px 24px 24px;
  outline: none;
  background-color: ",
              "#33cccc",
              ";
  color: ",
              "#fff",
              ";

  &:hover:not(.btn-default):not(.btn-loading):not([disabled]) {
    background-color: #43e4e4;
  }
  &:active:not(.btn-default):not(.btn-loading):not([disabled]) {
    background-color: #2fbcbc;
  }

  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  letter-spacing: normal;

  &:disabled {
    background-color: ",
              "#cfdae1",
              ";
    &,
    & * {
      cursor: not-allowed;
    }
  }
",
            ],
          },
          "displayName": "Submit__Input",
          "render": [Function],
          "styledComponentId": "Submit__Input-sc-1h1p8pz-0",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      type="submit"
      value="This is text"
    >
      <input
        className="Submit__Input-sc-1h1p8pz-0 kMJEtZ"
        type="submit"
        value="This is text"
      />
    </StyledComponent>
  </Submit__Input>
</Submit>
`;

exports[`Expect <Submit> to render properly with empty 1`] = `
<Submit
  onChange={[Function]}
  value=""
>
  <Submit__Input
    type="submit"
    value=""
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Submit__Input-sc-1h1p8pz-0",
            "isStatic": true,
            "lastClassName": "kMJEtZ",
            "rules": Array [
              "
  min-width: 200px;
  width: 100%;
  height: 80px;
  border: 0;
  padding: 24px 16px 24px 24px;
  outline: none;
  background-color: ",
              "#33cccc",
              ";
  color: ",
              "#fff",
              ";

  &:hover:not(.btn-default):not(.btn-loading):not([disabled]) {
    background-color: #43e4e4;
  }
  &:active:not(.btn-default):not(.btn-loading):not([disabled]) {
    background-color: #2fbcbc;
  }

  font-size: 14px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  letter-spacing: normal;

  &:disabled {
    background-color: ",
              "#cfdae1",
              ";
    &,
    & * {
      cursor: not-allowed;
    }
  }
",
            ],
          },
          "displayName": "Submit__Input",
          "render": [Function],
          "styledComponentId": "Submit__Input-sc-1h1p8pz-0",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      type="submit"
      value=""
    >
      <input
        className="Submit__Input-sc-1h1p8pz-0 kMJEtZ"
        type="submit"
        value=""
      />
    </StyledComponent>
  </Submit__Input>
</Submit>
`;
